common/buildcraft/builders/tile/TileFiller.java

```diff
@@ -94,18 +95,20 @@ public IdAllocator getIdAllocator() {
     public boolean placedWithVolume = false;
     public boolean placedWithVolumeType = false;
 
-    public final FullStatement<IFillerPattern> pattern;
+    public final FullStatement<IFillerPattern> patternStatement;
     private FilledTemplate patternTemplate;
     private Template blueprintTemplate;
     private BuildingInfo buildingInfo;
 
     public TileFiller() {
-        pattern = new FullStatement<>(FillerType.INSTANCE, 4, this::onStatementChange);
+        patternStatement = new FullStatement<>(FillerType.INSTANCE, 4, this::onStatementChange);
         caps.addProvider(new MjCapabilityHelper(new MjBatteryReceiver(battery)));
         caps.addCapabilityInstance(TilesAPI.CAP_CONTROLLABLE, this, EnumPipePart.VALUES);
         // TODO: Add a way of handling any itemstack or block properly,
         // including valid placement locations and removal.
-        StackInsertionChecker checker = (slot, stack) -> stack.getItem() instanceof ItemBlock;
+        StackInsertionChecker checker = (slot, stack) -> {
+            return stack.getItem() instanceof ItemBlock || stack.getItem() instanceof ItemBlockSpecial;
+        };
```

Comment:
This doesn't include all needed item, check old code. And code is really big while it's possible to make it more compact and less redundant, like old one.